thought_processing/thought.py

```diff
@@ -1,67 +1,53 @@
-    
-    def get_dict_to_save(self):
-        row = {'Highlight': self.highlight,
-            'Title': self.title,
-            'Author':self.author,
-            'URL':self.url,
-            'Note':self.note,
-            'Location':self.location,
-            'Date':self.date}
-        return row
-    
-    def get_date(self):
+
+    highlight: str = None
+    title: str = None
+    author: str = None
+    url: str = None
+    note: str = None
+    location: str = None
+    date: str = None
+    text: str = None
+
+    def _extract_date(self):
         regex = r"\d{4}-\d{2}-\d{2} \d{2}:\d{2}"
         match = re.search(regex, self.text)
         indexes = match.span()
-        date_text = self.text[indexes[0]:indexes[1]]
-        self.text = self.text.replace(date_text,'')
-        self.date = match[0].replace('-','.')
-        return self.date, self.text
+        date_text = self.text[indexes[0] : indexes[1]]
+        self.text = self.text.replace(date_text, "")
+        self.date = match[0].replace("-", ".")
 
-    def get_location(self):
+    def _extract_location(self):
         regex = r"Page No\.:\s*(\d+)"
         match = re.search(regex, self.text)
         self.location = match.group(1)
-        page_text = self.text[match.start():match.end()]
-        self.text = self.text.replace(page_text,'')
+        page_text = self.text[match.start() : match.end()]
+        self.text = self.text.replace(page_text, "")
         # print(text)
         # print(page_number)
         # row['Location']=page_number
-        self.text = self.text.replace('\xa0\xa0|\xa0\xa0','')
-        return self.location, self.text
+        self.text = self.text.replace("\xa0\xa0|\xa0\xa0", "")
 
-    def get_highlight_and_note(self):
-        ## be careful the text is already process and date and page number is removed,
-        ## must run date and location before
-        highlight_dirty=self.text
-        start_Note = highlight_dirty.find('【Note】')
-        self.highlight = highlight_dirty[:start_Note-1]
-        self.note = highlight_dirty[start_Note+6:-1]
-        self.note = self.note.replace(';',',')
+    def _extract_highlight_and_note(self):
+        # be careful the text is already process and date and page number is removed,
+        # must run date and location before
+        highlight_dirty = self.text
+        start_Note = highlight_dirty.find("【Note】")
+        self.highlight = highlight_dirty[: start_Note - 1]
+        self.note = highlight_dirty[start_Note + 6 : -1]
+        self.note = self.note.replace(";", ",")
         self.highlight = re.sub(r"^[\n\r]+", "", self.highlight)
-        self.highlight = self.highlight.replace('\n','')
-        self.highlight = self.highlight.replace(';',',')
-
-        return self.highlight, self.note, self.text
+        self.highlight = self.highlight.replace("\n", "")
+        self.highlight = self.highlight.replace(";", ",")
 
-    def parse_thought(self):
+    def extract_information(self):
```

Comment:
super dzięki, nie wiedziałem że tak można, doczytam o tym.

Reply:
This is done so that in files processing i can create the WiseThought and dump the dictionary of headers. if i have post init then this fails cause WiseThought is still empty and is unable to extract information - so its not always true that when you create WiseThought you always extract information. Not sure how to handle it otherwise since i want it to be automatic to get the headers and create the file before processing files.